libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp

```diff
@@ -122,8 +123,8 @@ void AP_InertialSensor_Backend::_notify_new_gyro_raw_sample(uint8_t instance,
         uint64_t now = AP_HAL::micros64();
         struct log_GYRO pkt = {
             LOG_PACKET_HEADER_INIT((uint8_t)(LOG_GYR1_MSG+instance)),
-            time_us   : now,
-            sample_us : sample_us?sample_us:now,
+            time_us   : sample_us?sample_us:now,
```

Comment:
I think the issue of timestamps merits more thought, and clearer definition. Perhaps I've made unwarranted assumptions or missed some documentation too?

For raw sensor data, I would argue that the correct value for the timestamp is the time at which the data was collected, not the time at which it was logged. Of course we don't have that, since we are polling the sensor, and that's why I wanted a sample counter in the first place, plus some assurance that I could detect gaps in the sample sequence.

Is there an expectation that the "timestamp" in a log message is just the time at which it was written to the log, or is it expected to convey the time at which the data in the log was measured?